### PR TITLE
Added root --version feature

### PR DESCRIPTION
**Pull request by AliAyub007**
**Changes:** 
Added feature to get root version using (root --version) command.

**Screenshot:** 

root version

### CODE REVIEW

File: rootx/src/rootx.cxx [hunk 1]
```diff
@@ -472,6 +471,26 @@ int main(int argc, char **argv)
          PrintUsage(argv[0]);
          return 1;
       }
+      if (!strncmp(argv[i], "--version", 9) || !strncmp(argv[i], "-version", 8) || !strncmp(argv[i], "--v", 3) || !strncmp(argv[i], "-v", 2)){
+         FILE *file;
+         char root_version[10];
+         
+         file = popen("root-config --version", "r");
```
Comment (Axel-Naumann) [inline rootx/src/rootx.cxx]: Call `gROOT->GetVersion()` instead?

File: rootx/src/rootx.cxx [hunk 2]
```diff
@@ -472,6 +471,26 @@ int main(int argc, char **argv)
          PrintUsage(argv[0]);
          return 1;
       }
+      if (!strncmp(argv[i], "--version", 9) || !strncmp(argv[i], "-version", 8) || !strncmp(argv[i], "--v", 3) || !strncmp(argv[i], "-v", 2)){
```
Comment (Axel-Naumann) [inline rootx/src/rootx.cxx]: We should only look for `"--version"`, please.

File: rootx/src/rootx.cxx (hunk 1 above)
Comment (AliAyub007) [reply rootx/src/rootx.cxx]: When I am trying to get the version of root using gROOT->GetVersion(), I am getting following errors: 

This is when I used #define gROOT (ROOT::GetROOT()) to get gROOT:
[ 72%] Built target rootcling_stage1
/Users/aliayubkhan/Documents/root/rootx/src/rootx.cxx:475:36: error: no member
      named 'GetROOT' in namespace 'ROOT'
          printf("Root Version: ", gROOT->GetVersion() + "\n");
                                   ^~~~~
/Users/aliayubkhan/Documents/root/rootx/src/rootx.cxx:442:22: note: expanded
      from macro 'gROOT'
#define gROOT (ROOT::GetROOT())
               ~~~~~~^
1 error generated.


I also tried by including TROOT.h but then I got this issue: 
/Users/aliayubkhan/Documents/root/rootx/src/rootx.cxx:476:36: warning: data
      argument not used by format string [-Wformat-extra-args]
          printf("Root Version: ", gROOT->GetVersion());
                 ~~~~~~~~~~~~~~~~  ^
/Users/aliayubkhan/Documents/build/include/TROOT.h:410:15: note: expanded from
      macro 'gROOT'
#define gROOT (ROOT::GetROOT())
              ^
[ 73%] Built target Base
[ 73%] Built target gitcommit
[ 73%] Built target rootnb.exe
[ 73%] Built target JupyROOT
[ 73%] Linking CXX executable ../bin/root
[ 73%] Built target complexDict
[ 73%] Built target Rint
[ 73%] Built target ThreadObjs
[ 73%] Built target root.exe
[ 73%] Built target Thread
[ 73%] Built target G__Imt
[ 73%] Built target G__RIO
Undefined symbols for architecture x86_64:
  "TVersionCheck::TVersionCheck(int)", referenced from:
      _GLOBAL_sub_I_rootx.cxx in rootx.cxx.o
  "ROOT::GetROOT()", referenced from:
      _main in rootx.cxx.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/root] Error 1
make[1]: *** [rootx/CMakeFiles/root.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 73%] Built target Imt
[ 75%] Built target RIOObjs
make: *** [all] Error 2

File: rootx/src/rootx.cxx (hunk 2 above)
Comment (AliAyub007) [reply rootx/src/rootx.cxx]: Sure removing others.

File: rootx/src/rootx.cxx (hunk 1 above)
Comment (Axel-Naumann) [reply rootx/src/rootx.cxx]: Check your use of `printf`...

Comment (AliAyub007) [reply rootx/src/rootx.cxx]: Oh I have copied wrong logs here.  I am getting this error only on invoking gROOT->GetVersion()  

[ 73%] Built target map2Dict
[ 73%] Built target vectorDict
Undefined symbols for architecture x86_64:
  "TVersionCheck::TVersionCheck(int)", referenced from:
      __GLOBAL__sub_I_rootx.cxx in rootx.cxx.o
  "ROOT::GetROOT()", referenced from:
      _main in rootx.cxx.o
[ 73%] Built target unordered_setDict
[ 73%] Built target mapDict
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[ 73%] Built target unordered_multisetDict
make[2]: *** [bin/root] Error 1
make[1]: *** [rootx/CMakeFiles/root.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 73%] Built target forward_listDict
[ 73%] Built target dequeDict
[ 73%] Built target New
[ 73%] Built target unordered_multimapDict
make: *** [all] Error 2

**Comment by Axel-Naumann** [comment]
Ha, thanks, indeed. Could you move the handling of `--version` into `TApplication::GetOptions` (see core/base/TApplication.cxx)

**Comment by AliAyub007** [comment]
Added --version to TApplication::GetOptions and removed all other variants.

**Comment by Axel-Naumann** [comment]
Could you fix the capitalization of "ROOT" in the print out? Please amend these commits, no need to keep them separate. Thanks!

**Comment by Axel-Naumann** [comment]
O, and can you include the git commit hash and architecture? The commit you can get from gROOT. I don't know where to get the architecture from the top of my head, but check where ROOT prints the banner (TRint.cxx), it shows it, too. Thanks!

**Comment by AliAyub007** [comment]
@Axel-Naumann Sure I will do that. I will squash the commits when work will be complete and good to get merged. Can you kindly let me know if I should show it separately from version in next line? Thanks!

**Comment by Axel-Naumann** [comment]
Separate lines might be nicer for tooling (e.g. grepping the git commit from `root --version`. Thanks!

**Comment by AliAyub007** [comment]
screen shot 2018-09-18 at 3 58 17 pm

@Axel-Naumann Kindly see this attached screenshot. If this is good to go I will commit the code.

**Comment by Axel-Naumann** [comment]
Awesome! The date is the commit date or the build date? If commit date: perfect. If build date, consider moving it up one line, i.e. `Built for macosx64 on Sept 18,...`

Thanks!

**Comment by AliAyub007** [comment]
@Axel-Naumann It was built time. So I made it as you suggested. 

screen shot 2018-09-18 at 10 23 37 pm

This is same in banner as well, means it is built time in banner which is written with commit details. Should I update it here or it should be done separately? Or I should just not do that now? Thanks!

**Comment by Axel-Naumann** [comment]
This is perfect! Please apply these changes here, either rewriting your branch history or adding to it (I will squash on merge). Thanks & congrats!

**Comment by AliAyub007** [comment]
Updated the banner. Thanks!! Looking forward to contribute more.

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

**Pull request merged**